modules/shared/src/main/scala/io/constellationnetwork/security/Hasher.scala

```diff
@@ -88,7 +91,49 @@ object Hasher {
       hashKryo(data)
   }
 
-  def forJson[F[_]: Sync: JsonSerializer]: Hasher[F] = new Hasher[F] {
+  def forJson[F[_]: Sync: JsonSerializer]: Hasher[F] =
+    sys.env.get("CL_CACHE_JSON_HASH").exists(_.toLowerCase == "true") match {
+      case true  => forJsonCached[F]
+      case false => forJsonUncached[F]
+    }
+
+  def forJsonCached[F[_]: Sync: JsonSerializer]: Hasher[F] = {
+    val cache: Cache[Any, Hash] = Scaffeine()
+      .recordStats()
+      .expireAfterAccess(5.minutes)
+      .maximumSize(10000)
```

Comment:
Both of these should maybe be configurations, but not necessary, could be higher potentially but this is safe.